azurerm/resource_arm_api_management_subscription.go

```diff
@@ -118,12 +118,15 @@ func resourceArmApiManagementSubscriptionCreateUpdate(d *schema.ResourceData, me
 	state := d.Get("state").(string)
 	userId := d.Get("user_id").(string)
 
+	scope := "/products/" + productId
+	ownerId := "/users/" + userId
+	// TODO: VERIFY THAT THIS WORKS AS EXPECTED
 	params := apimanagement.SubscriptionCreateParameters{
 		SubscriptionCreateParameterProperties: &apimanagement.SubscriptionCreateParameterProperties{
 			DisplayName: utils.String(displayName),
-			ProductID:   utils.String(productId),
+			Scope:       utils.String(scope),
 			State:       apimanagement.SubscriptionState(state),
-			UserID:      utils.String(userId),
+			OwnerID:     utils.String(ownerId),
```

Comment:
Major breaking change in the Azure SDK. 
`ProductID -> Scope`
`UserID -> OwnerID`
The values have different formats. Example from the docs:
```
{
  "properties": {
    "ownerId": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/users/57127d485157a511ace86ae7",
    "scope": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/products/5600b59475ff190048060002",
    "displayName": "testsub"
  }
}
```

The Go SDK looked like it supported relative paths, i.e. `/users/USER-ID` 
The _scope_ property supports other things than the product, but I used product since it was used in the current implementation. 

https://docs.microsoft.com/en-us/rest/api/apimanagement/2019-01-01/subscription/createorupdate